lib/www.scm

```diff
@@ -246,6 +246,74 @@
              (redirect-list)))
      ,@(html-body-as-sxml "doc/redirect.html"))))
 
+(define (project-https-uri project)
+  (string-append "https://" (get-string 'project-id project) ".scheme.org/"))
+
+(define (project-source-uri project)
+  (let ((source (get-list 'source project)))
+    (and (not (null? source))
+         (let ((host (get-string 'host source))
+               (organization (get-string 'organization source))
+               (repository (get-string 'repository source))
+               (path (get-string? 'path source)))
+           (string-append "https://" host "/" organization "/" repository
+                          (if path
+                              (string-append "/tree/master/" path)
+                              ""))))))
+
+(define (project-server project)
+  (define (dotted? host) (memv #\. (string->list host)))
+  (any (lambda (record)
+         (and (= (length record) 2)
+              (eqv? 'CNAME (first record))
+              (let ((host (second record)))
+                (and (not (dotted? host))
+                     host))))
+       (get-list 'dns project)))
+
+(define (source-project-tds project)
+  `((td (a (@ (href ,(project-https-uri project)))
+           (code ,(get-string 'project-id project))))
+    (td ,(or (project-server project)
+             ""))
+    (td ,(cond ((get-string? 'redirect project)
+                "redirect")
+               ((get-symbol-boolean? 'dynamic? project)
+                "dynamic")
+               (else
+                "static"))
+        ,@(if (get-symbol-boolean 'display? project) '() '(" hidden")))
+    (td ,(let ((uri (project-source-uri project)))
+           (if uri
+               `(a (@ (href ,uri))
+                   (img (@ (alt "GitHub")
+                           (src "github16.png"))))
+               "")))))
+
+(define (write-source-page)
+  (write-html-file
+   "www/scheme.org/source/index.html"
+   "Scheme.org Source"
+   "Explains where to find the source code of Scheme.org."
+   `((h1 "Scheme.org Source")
+     (table
+      (@ (class "full-width no-border"))
+      ,@(append-map
+         (lambda (group)
+           `((tr (td (@ (colspan "4")
+                        (class "no-border"))
+                     (h2 ,(project-group-title group))))
+             (tr (th "Subdomain")
+                 (th "Server")
+                 (th "Type")
+                 (th "Source"))
+             ,@(map (let ((even-row? (circular-generator #t #f)))
```

Comment:
You can do this directly in CSS.

https://www.w3.org/Style/Examples/007/evenodd.en.html

Reply:
Interesting. I didn't know that.

A couple points against it:

* People on StackOverflow claim the CSS isn't supported on Internet Explorer. I assume it isn't supported on the lightweight Unix GUI browsers that some programmers are using either.

* In this case, the whole page is one big table. The `<h2>` headings are sunk into `<th>` elements. Each group under a heading has its own even/odd counter.

For these reasons, and since we generate the page anyway, I'd like to keep the manual CSS classes. But it would be good to expore the native CSS option in the future.

Reply:
We shouldn't worry about MSIE.  We're long past the time when it was important.  Modern browsers, even from Microsoft, have all of this stuff.  And having even/odd highlighting break (on MSIE) isn't a big loss.  People who are still running that will have much bigger problems to worry about.

I agree with your all-one-big-table argument.

Reply:
Mostly agree about IE. But there are a non-trivial amount of programmers who prefer a light browser, resorting to a heavy one as infrequently as they can manage. Most stuff on Scheme.org should work comfortably for them.